napari/layers/points/points.py

```diff
@@ -851,23 +853,37 @@ def current_size(self) -> Union[int, float]:
 
     @current_size.setter
     def current_size(self, size: Union[None, float]) -> None:
-        if (isinstance(size, numbers.Number) and size < 0) or (
-            isinstance(size, list) and min(size) < 0
-        ):
+        if isinstance(size, (list, tuple, np.ndarray)):
             warnings.warn(
-                message=trans._(
-                    'current_size value must be positive, value will be left at {value}.',
+                trans._(
+                    "Point sizes must be isotropic; the average from each dimension will be used instead. "
+                    "This will become an error in a future version.",
+                    deferred=True,
+                ),
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
+            size = size[-1]
+        if not isinstance(size, numbers.Number):
+            raise TypeError(
+                trans._(
+                    'currrent size must be a number',
+                    deferred=True,
+                )
+            )
+        if size < 0:
+            raise ValueError(
+                trans._(
+                    'current_size value must be positive.',
                     deferred=True,
-                    value=self.current_size,
                 ),
-                category=RuntimeWarning,
             )
-            size = self.current_size
+
         self._current_size = size
         if self._update_properties and len(self.selected_data) > 0:
-            for i in self.selected_data:
-                self.size[i, :] = (self.size[i, :] > 0) * size
-            self._clear_extent_augmented()
```

Comment:
As `Points._extent_augmented` is dependent on `Points.size` and you're doing an in-place assignment to `Points.size`, I don't think you should delete this right?

Reply:
good catch!